crates/astria-sequencer-relayer/src/types.rs

```diff
@@ -167,6 +168,12 @@ pub struct SequencerBlockData {
     pub rollup_txs: HashMap<Namespace, Vec<IndexedTransaction>>,
 }
 
+impl Hash for SequencerBlockData {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.block_hash.hash(state);
+    }
+}
```

Comment:
what is this? why is it re-hashing the block hash? I assume this is for the `PriorityQueue`, but should this just return `self.block_hash`?